clarinet/contracts/main-swapr.clar

```diff
@@ -285,97 +295,81 @@
-                      (get fee-balance-y pair)
-                    )
-                  ,
-                  fee-to-address: (get fee-to-address pair),
-                  name: (get name pair),
-                  swapr-token: (get swapr-token pair),
-                }
-              )
-              (ok (list dx dy))
-            )
-            transfer-failed-err
-          )
-          too-much-slippage-err
-        )
-      )
-    )
+  (let ((token-x (contract-of token-x-trait)) 
+        (token-y (contract-of token-y-trait))
+        (pair (unwrap! (map-get? pairs-data-map { token-x: token-x, token-y: token-y }) invalid-pair-err))
+        (balance-x (get balance-x pair))
+        (balance-y (get balance-y pair))
+        (contract-address (as-contract tx-sender))
+        (sender tx-sender)
+        ;; check formula, vs x-for-y???
+        (dx (/ (* u997 balance-x dy) (+ (* u1000 balance-y) (* u997 dy)))) ;; overall fee is 30 bp, either all for the pool, or 25 bp for pool and 5 bp for operator
+        (fee (/ (* u5 dy) u10000)) ;; 5 bp
+        (pair-updated (merge pair {
+          balance-x: (- (get balance-x pair) dx),
+          balance-y: (+ (get balance-y pair) dy),
+          fee-balance-y: (if (is-some (get fee-to-address pair))  ;; only collect fee when fee-to-address is set
+            (+ fee (get fee-balance-y pair))
+            (get fee-balance-y pair))
+        })))
+
+    (asserts! (< min-dx dx) too-much-slippage-err)
+
+    (asserts! 
+      (and
+        ;; TODO(psq): check that the amount transfered in matches the amount requested
+        (is-ok (as-contract (contract-call? token-x-trait transfer dx contract-address sender)))
+        (is-ok (contract-call? token-y-trait transfer dy sender contract-address)))
+      transfer-failed-err)
+
+    (map-set pairs-data-map { token-x: token-x, token-y: token-y } pair-updated)
+    (print { object: "pair", action: "swap-y-for-x", data: pair-updated })
+    (ok (list dx dy))
   )
 )
 
 ;; ;; activate the contract fee for swaps by setting the collection address, restricted to contract owner
 (define-public (set-fee-to-address (token-x principal) (token-y principal) (address principal))
   (let ((pair (unwrap! (map-get? pairs-data-map { token-x: token-x, token-y: token-y }) invalid-pair-err)))
-    (if (is-eq tx-sender contract-owner)
-      (begin
-        (map-set pairs-data-map { token-x: token-x, token-y: token-y }
-          {
-            shares-total: (get shares-total pair),
-            balance-x: (get balance-y pair),
-            balance-y: (get balance-y pair),
-            fee-balance-x: (get fee-balance-y pair),
-            fee-balance-y: (get fee-balance-y pair),
-            fee-to-address: (some address),
-            name: (get name pair),
-            swapr-token: (get swapr-token pair),
-          }
-        )
-        (ok true)
-      )
-      not-owner-err
+
+    (asserts! (is-eq tx-sender contract-owner) not-owner-err)
```

Comment:
yeah, using asserts simplifies the if nesting, should have done this a long time ago